ops/pebble.py

```diff
@@ -1926,3 +1910,188 @@ def _websocket_url(self, task_id: str, websocket_id: str) -> str:
         base_url = self.base_url.replace('http://', 'ws://')
         url = '{}/v1/tasks/{}/websocket/{}'.format(base_url, task_id, websocket_id)
         return url
+
+
+class StopParsing(Exception):   # noqa: N818
+    """Signals that the parser should stop parsing, at least for now."""
```

Comment:
Something maybe feels off about triggering on the generic Exception class here, but I can't think why atm. I don't think that whatever is bouncing around in the back of my brain is super important, though.

Reply:
Yeah - from a pure FSM perspective, it may be arguable that we should transition to a paused state and track where things were, then transition back upon new data being served.  The exception seemed a simple way to avoid having to do that extra tracking.  I'd be more against it if the FSM in question were likely to grow more over time, but the scope here is relatively limited.

I'll try to come up with a way that avoids the exception; it may be easier than I could see at the time.

Reply:
OK, maybe this wasn't too bad.  Removed the exception, and also tried to remove multiple return paths at the cost of more explicit if/else logic.  I'm not sure that I fully like the latter part of that, but feedback (or proposed alternatives) are welcome.